### PR TITLE
Add Helm chart and provisioning scripts

### PR DESCRIPTION
**Pull request by kim-cc**
### Description
Add Helm chart, provisioning scripts and documentation on how to use it.

### Relevant Issues/PRs
Closes DEV-142

### Type(s) of Change (Split the PR if you check many)
- [ ] Added user feature
- [ ] Added technical feature
- [ ] Added tests
- [ ] Refactor
- [ ] Bugfix
- [X] Updated documentation
- [ ] Kubernetes changes
- [ ] CI/CD workflow changes

### Testing Checklist
- [ ] I have added/updated unit tests for functional changes.
- [ ] I have added/updated integration tests for changes that affect dependent systems.
- [ ] I have added/updated end-to-end tests for feature changes.

### General Checklist
- [X] I have pulled in the latest changes from master.
- [ ] I have run `make lint`.
- [ ] I have successfully run `make tests`.
- [ ] I have updated relevant CI/CD workflows.
- [ ] I have updated relevant Kubernetes manifests/scripts.
- [ ] I have updated/added relevant (internal and external) documentation (readme, doc comments, etc).
- [ ] I have updated the dependency map to reflect my changes.
- [ ] I have added the license to new source code files.
- [X] I have spent some time looking over the full diff before creating this PR.

### CODE REVIEW

**Review by philip-cc — APPROVED** [review]
Seems to work.

File: e2etests/makefile [hunk 1]
```diff
@@ -28,14 +28,30 @@ lint: ## Lint the codebase
...
-	go mod tidy
 	USER_INFO=$$(docker exec encryptonize-objects /encryptonize-objects create-user rcudiom  | tail -n 1) && \
 		export E2E_TEST_UID=$$(echo $$USER_INFO | jq -r ".user_id") && \
 		export E2E_TEST_PASS=$$(echo $$USER_INFO | jq -r ".password") && \
 		go test -count=1 -v ./...
 	$(MAKE) -C .. docker-down
+
+.PHONY: tests-kubernetes
+tests-kubernetes: POD_NAME = $$(kubectl get pods -l app.kubernetes.io/name=encryptonize-objects | tail -n 1 | awk '{print $$1}')
+tests-kubernetes: PORT_FORWARD_CMD = kubectl port-forward ${POD_NAME} 9000:9000
```
Comment (philip-cc) [inline e2etests/makefile:48]: You could just do this on the deployment, so you don't need the fetch the pod name above.

Comment (kim-cc) [reply e2etests/makefile:48]: The pod name is also needed on line 53 or do I misunderstand you comment?

Comment (philip-cc) [reply e2etests/makefile:48]: You can also do `exec` on a deployment, at least according to `kubectl exec --help`.

**Review by philip-cc — APPROVED** [review]

**Pull request merged**